Path Navigator Issue with Loading State;

aachavarria opened the pull request:
Pending issue with the tooltip

### Ticket reference or full description of what's in the PR

aachavarria committed:
adding isFetching to PathNavigator.tsx;
WIP issue with tooltip;

aachavarria committed:
ItemMenu issues with PathNavigator.tsx;
mimeType Fix;

aachavarria committed:
PathNavigatorUI.tsx: Dynamic Loader Items;
Removing parent from GetChildrenResponse.ts;
Creating ItemWithChildren Interface;
Creating fetchItemWithPath() and fetchItemsByPath();
Updating pathNavigator.ts to use fetchItemsByPath and fetchItemWithPath;
Fixes on getIndividualPaths to avoid duplicates;

rart commented on ui/app/src/services/content.ts:
```diff
@@ -810,6 +806,45 @@ export function getChildrenByPath(
+  return forkJoin(requests);
+}
+
+export function fetchItemWithPath(
```
fetchItemWithChildrenByPath

rart commented on ui/app/src/state/actions/pathNavigator.ts:
```diff
@@ -46,7 +46,7 @@ export const pathNavigatorConditionallySetPath = createAction<PayloadWithId<{ pa
 
 export const pathNavigatorConditionallySetPathComplete = createAction<
-  PayloadWithId<{ path: string; response: GetChildrenResponse }>
+  PayloadWithId<{ path: string; item?: DetailedItem; children: GetChildrenResponse }>
```
Is this item ever **not** the parent? Naming it parent would be clearer.

rart commented on ui/app/src/modules/Content/Authoring/SingleItemSelector.tsx:
```diff
@@ -305,9 +276,13 @@ export const fetchChildrenByPath = createAction<string>('FETCH_CHILDREN_BY_PATH'
-export const fetchChildrenByPathComplete = createAction<GetChildrenResponse>('FETCH_CHILDREN_BY_PATH_COMPLETE');
+export const fetchChildrenByPathComplete = createAction<{ item?: DetailedItem; children: GetChildrenResponse }>(
+  'FETCH_CHILDREN_BY_PATH_COMPLETE'
+);
```
What are these exported for?

rart commented on ui/app/src/models/Item.ts:
```diff
@@ -107,3 +109,8 @@ export interface PasteItem {
   children?: PasteItem[];
 }
+
+export interface ItemWithChildren {
```
==> `GetItemWithChildrenResponse`. Move to it's own file like GetChildrenResponse

aachavarria committed:
Merging Changes from JWT;
Renaming and moving ItemWithChildren to GetItemWithChildrenResponse;
Renaming fetchItemWithPath to fetchItemWithChildrenByPath;

aachavarria committed:
adding 'renderingTemplate'

aachavarria committed:
Fixing issue with SingleItemSelector.tsx to use the parentPath of the selected item on opening;

aachavarria committed:
adding isFetching to pathNavigatorRefresh and pathNavigatorSetKeyword

rart commented on ui/app/src/components/Navigation/PathNavigator/utils.ts:
```diff
@@ -25,6 +25,7 @@ export function isPreviewable(item: DetailedItem): boolean {
     item.systemType === 'component' ||
     item.systemType === 'asset' ||
     item.systemType === 'template' ||
```
x

rart commented on ui/app/src/utils/itemActions.ts:
```diff
@@ -293,6 +293,7 @@ export function generateSingleItemOptions(item: DetailedItem, permissions: Looku
     case 'taxonomy':
     case 'component':
+    case 'renderingTemplate':
     case 'template':
```
x

rart commented on ui/app/src/components/Navigation/PathNavigator/PathNavigator.tsx:
```diff
@@ -306,7 +306,7 @@ export default function PathNavigator(props: PathNavigatorProps) {
         let mode = 'txt';
 
-        if (item.systemType === 'template') {
+        if (item.systemType === 'template' || item.systemType === 'renderingTemplate') {
```
x 'template'

aachavarria committed:
removing using of 'template' system Type

rart commented on ui/app/src/utils/content.ts:
```diff
@@ -123,7 +123,7 @@ export function getSystemTypeFromPath(path: string): string {
 function getLegacyItemSystemType(item: LegacyItem) {
```
Set the return type to the string list of system types

aachavarria committed:
Creating SystemType Type;

aachavarria committed:
adding SystemType to BaseItem;

aachavarria committed:
Fix on Search Items Number
